### PR TITLE
JavaScript API documentation improvements and more

### PR DESCRIPTION
**Pull request by mzabaluev**
Added and improved method docs.
Documented some of the types defined in wallet-js API.
Also updated the docs in wallet-core to match.

### CODE REVIEW

**Comment by mzabaluev** [comment]
The lint failure should be addressed in #168.

File: bindings/wallet-js/src/lib.rs [hunk 1]
```diff
@@ -68,21 +77,26 @@ pub struct BlockDate(chain_impl_mockchain::block::BlockDate);
...
-    /// this function will work for all yoroi, daedalus and other wallets
-    /// as it will try every kind of wallet anyway
+    /// Recovers a wallet from the given BIP39 mnemonics and password.
     ///
     /// You can also use this function to recover a wallet even after you have
-    /// transferred all the funds to the new format (see the _convert_ function)
+    /// transferred all the funds to the new format (see the `convert` method).
     ///
-    /// the mnemonics should be in english
+    /// The mnemonics should be in English, or normalized using NFKD.
```
Comment (enzoc4) [inline bindings/wallet-js/src/lib.rs:85]: I don't think this is an **or** but an **and**. The character encoding shouldn't have anything to do with the language. We have all the dictionaries, but we are only checking with the english one.

File: bindings/wallet-js/src/lib.rs [hunk 2]
```diff
@@ -340,15 +387,17 @@ macro_rules! impl_secret_key {
 
         #[wasm_bindgen]
         impl $name {
+            /// Generates the key using OS-provided entropy.
```
Comment (enzoc4) [inline bindings/wallet-js/src/lib.rs:390]: Maybe it's worth pointing out to the specific apis that `OsRng` uses? given that this is a wasm/js specific crate they should be only browser and node.
Not really sure if a web developer would find that useful though.

File: bindings/wallet-core/src/wallet.rs [hunk 3]
```diff
@@ -188,10 +192,10 @@ impl Wallet {
         self.account.confirm(&id);
     }
 
-    /// get access to all the pending transaction
+    /// Get IDs of all pending transactions produced by `convert`.
```
Comment (enzoc4) [inline bindings/wallet-core/src/wallet.rs:195]: this technically (should at least) return *all* the pending transactions. The comment in the js api is correct though.

**Review by enzoc4 — APPROVED** [review]

File: bindings/wallet-core/src/wallet.rs (hunk 3 above)
Comment (mzabaluev) [reply bindings/wallet-core/src/wallet.rs:195]: The initial transactions in block0 are immediately confirmed, where do the others come from?

**Comment by mzabaluev** [comment]
Test failures should be fixed by #173

**Pull request merged**